### PR TITLE
Mock auth requests

### PR DESCRIPTION
**Pull request by jayrbolton**
This gets CI working and removes dependency on the live auth server for testing.

I'm using this small docker mock service that I wrote:
https://github.com/jayrbolton/mock_microservice

Issue #12

### CODE REVIEW

**Comment by jayrbolton** [comment]
Still debuggin the Travis run here...

File: src/relation_engine_server/bulk_import.py [hunk 1]
```diff
@@ -33,3 +35,17 @@ def _write_edge_key(json_line):
...
+    with open(file_path, 'rb') as file_desc:
+        resp = requests.post(
+            db_url + '/_api/import',
+            data=file_desc,
+            auth=(db_user, db_pass),
+            params=query
+        )
+    if not resp.ok:
+        raise ArangoServerError(resp.text)
+    return resp.text
```
Comment (JamesJeffryes) [inline src/relation_engine_server/bulk_import.py]: I could make a case for why we'd want this function here or in arango_cilent (though I probably lean towards the latter) but if we do move it here, I think you should remove the bulk_import from arango_cilent

File: src/test/test_api.py [hunk 2]
```diff
@@ -38,68 +45,94 @@ def save_test_docs(count, edges=False):
...
+        resp = requests.put(
+            url + '/api/documents?on_duplicate=error&overwrite=true&collection'
+        ).json()
+        self.assertEqual(resp['error'], 'Missing header: Authorization')
+
+    def test_save_documents_invalid_auth(self):
+        """Test an invalid attempt to save a doc with a bad auth token."""
         resp = requests.put(
             url + '/api/documents?on_duplicate=error&overwrite=true&collection',
             headers={'Authorization': 'Bearer xyz'}
```
Comment (JamesJeffryes) [inline src/test/test_api.py]: Pedantic, but if you define an 'invalid_token' variable, I'd expect it to be used here

File: src/test/test_api.py [hunk 3]
```diff
@@ -213,12 +246,25 @@ def test_admin_query(self):
...
+            })
+        ).json()
+        self.assertEqual(resp['error'], '403 - Unauthorized')
+
     def test_admin_query_invalid_auth(self):
         """Test the error response for an ad-hoc admin query without auth."""
         resp = requests.post(
             url + '/api/query_results',
             params={},
-            headers={'Content-Type': 'application/json', 'Authorization': 'xyz'},
```
Comment (JamesJeffryes) [inline src/test/test_api.py]: And here

**Comment by jayrbolton** [comment]
Thanks , good catches as usual. I moved that function out because I found
it weird to have a function called bulk import and a separate module called
bulk import. I’ll try to rearrange things

**Comment by jayrbolton** [comment]
- Reorganized the bulk import functions, removed accidental dupe code
- Using invalid_token in the tests

**Comment by jayrbolton** [comment]
Dammit travis.

**Comment by jayrbolton** [comment]
We're hitting the github API rate limits on travis. This rate limit is only for fetching the latest release of the specs. I think you can download the release gzip without limits. I'm thinking we can just put the url of the latest spec release in the environment and use that as a fallback.

**Pull request merged**